org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/reconciler/XtextDocumentReconcileStrategy.java

```diff
@@ -108,7 +108,7 @@ public void setEditor(XtextEditor editor) {
 	@Override
 	public void setDocument(IDocument document) {
 		if (!(document instanceof XtextDocument)) {
-			throw new IllegalArgumentException("Document must be an " + XtextDocument.class.getSimpleName());
+			throw new IllegalArgumentException("Document must be an " + XtextDocument.class + " but was " + document.getClass().getName());
```

Comment:
a~~n~~

Reply:
It's the sound that matters and I'd pronounce it as `ExtextDocument`.

cf https://english.stackexchange.com/questions/8181/does-one-use-a-or-an-before-the-word-x-ray